src/editortoolkit.cpp

```diff
@@ -625,6 +639,139 @@ std::string EditorToolkit::EditInfo()
     return m_editInfo;
 }
 
+bool EditorToolkit::Group(std::string groupType, std::vector<std::string> elementIds)
+{
+    Object *newParent, *neumeParent, *newSylParent, *sylParent;
+    std::set<Object *> parents;
+
+    if (!m_doc->GetDrawingPage()) {
+        LogError("Could not get the drawing page.");
+        return false;
+    }
+
+    if(groupType.compare("nc") == 0){
+        for (auto it = elementIds.begin(); it != elementIds.end(); ++it) {
+            Object *el = m_doc->GetDrawingPage()->FindChildByUuid(*it);
+            if (elementIds.begin() == it){
+                newParent = el->GetFirstParent(NEUME);
+                assert(newParent);
+                newSylParent = newParent->GetFirstParent(SYLLABLE);
+                assert(newSylParent);
+            }
+            else {
+                if(newParent && newSylParent){
+                    neumeParent = el->GetFirstParent(NEUME);
+                    assert(neumeParent);
+
+                    el->MoveItselfTo(newParent);
+
+                    sylParent = neumeParent->GetFirstParent(SYLLABLE);
+                    assert(sylParent);
+                
+                    std::string className = sylParent->GetClassName();
+                    if(className.compare("Syllable") != 0) return false;
+
+                    if(sylParent != newSylParent){
+                        parents.insert(sylParent);
+                    }   
+                } 
+            }
+        }
+    }
+    else if (groupType.compare("neume") == 0){
+        for (auto it = elementIds.begin(); it != elementIds.end(); ++it) {
+            Object *el = m_doc->GetDrawingPage()->FindChildByUuid(*it);
+            if (elementIds.begin() == it){
+                newParent = el->GetFirstParent(SYLLABLE);
+                assert(newParent);
+            }
+            else {
+                if(newParent){
+                    sylParent = el->GetFirstParent(SYLLABLE);
+                    assert(sylParent);
+
+                    el->MoveItselfTo(newParent);
+                
+                    std::string className = sylParent->GetClassName();
+                    if(className.compare("Syllable") != 0) return false;
+
+                    if(sylParent != newParent){
+                        parents.insert(sylParent);
+                    }   
+                } 
+            }
+        }
+    }
+    else{
+        return false;
```

Comment:
Include some `LogWarning` or `LogError` statements. There are many situations where this can return false and the user won't have any idea why.